doc/release/upcoming_changes/27357.improvement.rst

```diff
@@ -1,3 +1,3 @@
-* The ``__repr__`` for :ref:`_user.user-defined-data-types` now prefers the
```

Comment:
`_user.user-defined-data-types` is not something we should put here literally. We should either update the ref to the correct link, or just say `User-defined data types` in plain text.

Reply:
My intent was to link to the labeled section in this doc: https://github.com/numpy/numpy/blob/760dbe9e65244c8e42ad41949b08b3da0ea0712b/doc/source/user/c-info.beyond-basics.rst#L206

Reply:
By and large, references in release notes are iffy. Fix it as you see fit.

Reply:
Maybe, but the release notes also need to usable as stand alone documents, I need to undo all the references when I post them on github for the release.